testing/simapp/app.go

```diff
@@ -421,6 +423,15 @@ func NewSimApp(
 	ibcRouter := porttypes.NewRouter()
 	ibcAppRouter := porttypes.NewAppRouter()
 
+	// add multiple a mock IBCModuleV2 modules whose functionality can be overridden in tests.
+	mockModuleV2A := ibcmock.NewIBCModuleV2(ibcmock.NewIBCV2App())
+	ibcAppRouter.AddV2Route(ibcmock.ModuleNameV2+"A", mockModuleV2A)
+	app.MockV2ModuleA = mockModuleV2A
+
+	mockModuleV2B := ibcmock.NewIBCModuleV2(ibcmock.NewIBCV2App())
+	ibcAppRouter.AddV2Route(ibcmock.ModuleNameV2+"B", mockModuleV2B)
+	app.MockV2ModuleB = mockModuleV2B
```

Comment:
we could probably design a mock module that is a single module, but allows for dynamic modification of behaviour based on payload/version provided etc.

If anyone feels like we should do that now, happy to give it a go, otherwise can just use this dumber approach of having 2 instances.